### PR TITLE
225 crud polling stations

### LINKED ISSUE #255: [Polling Stations] Add CRUD endpoints for managing polling stations
Information about polling stations should be displayed in the NGO admin panel. It should also be editable.

:exclamation: do not delete old endpoints in order to keep backwards compatibility with older client app versions

Endpoints to be added:

- [ ] Add endpoint to view all the polling stations
`GET api/v2/polling-station`
- [ ] Add endpoint to edit a polling station _- action only allowed for to admin account (an ngo admin belonging to an ngo that has the organizer flag set to true)_
`POST api/v2/polling-station/{id}`

Update polling-station-info related endpoints (also keep the old ones for backwards compatibility):
- [ ] Add endpoint to post polling station infos (copy functionality from `POST api/v1/polling-station`)
`POST api/v2/polling-station-info`
- [ ] Add endpoint to put polling station infos (copy functionality from`PUT api/v1/polling-station`)
`PUT api/v2/polling-station-info`

### PR DESCRIPTION
**Pull request by decembrya**
Closes #255 
https://github.com/code4romania/monitorizare-vot/issues/255

### CODE REVIEW

File: src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpGet]
+        [Produces(typeof(IEnumerable<GetPollingStation>))]
+        public async Task<IActionResult> GetAllPollingStations([FromQuery]PollingStationsFilter pollingStationsFilter)
+        {
+            //todo fix paging model defaults
```
Comment (aniri) [inline src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs]: Please fix this :smile: It crashes with 500 now if the pagination params are missing from the request (`Page=1&PageSize=1`). It should the defaults in this case :smile:

File: src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            var request = _mapper.Map<GetPollingStations>(pollingStationsFilter);
+
+            var result = await _mediator.Send(request);
+
+            return Ok(result);
+        }
+
+        [HttpPut("{id}")]
+        [Authorize("NgoAdmin")]
```
Comment (aniri) [inline src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs]: Please use `[Authorize("Organizer")]` here to restrict this to master admins :smile:

File: src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs (hunk 1 above)
Comment (decembrya) [reply src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs]: Ah yeah will do, sorry I noticed it but forgot to fix it

File: src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs (hunk 2 above)
Comment (decembrya) [reply src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs]: Done

File: src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs (hunk 1 above)
Comment (decembrya) [reply src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs]: Should be ok now :D

**Comment by aniri** [comment]
@bvizureanu tested this and works as expected. can you also add you review? :D

**Review by aniri — DISMISSED** [review]

**Review by aniri — APPROVED** [review]

**Pull request merged**